cfme/tests/cloud/test_providers.py

```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
     """
-    pass
+    # preparations for test
+    request.addfinalizer(lambda: provider.delete_if_exists())
+    topic = provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic')
+    if topic:
+        provider.mgmt.delete_topic(topic)
+
+    # SNS topic should be automatically created during provider creation in CFME
+    provider.create()
+    new_topic = wait_for(lambda: provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic'), delay=15,
```

Comment:
Note you don't need the lambda here as well as you can tell `wait_for` what arguments to use but yeah who does remember all the arguments for the `wait_for` .. (:

Reply:
We have a bunch of places where this is true, I've tried to push towards using func_args more :)